### PR TITLE
feat(fileViewer) MAG-203 : now file viewer to display files in magnets

### PR DESCRIPTION
**Pull request by Rseuret**
## Describe your changes
using file viewer to display files in some elements 

## Issue ticket number and link
https://jira.support-ent.fr/browse/MAG-203

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/directives/file-viewer/csv-viewer/csv-viewer.directive.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
+import { ng } from 'entcore';
```
Comment (Rseuret) [inline src/main/resources/public/ts/directives/file-viewer/csv-viewer/csv-viewer.directive.ts:1]: directive d ode c est pas ouf mais y a pas le tps

File: src/main/resources/public/ts/directives/file-viewer/csv-viewer/csv-viewer.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<div>
```
Comment (Rseuret) [inline src/main/resources/public/ts/directives/file-viewer/csv-viewer/csv-viewer.html:1]: fichier pour plus tard car pasl e temps

File: src/main/resources/public/ts/directives/file-viewer/file-viewer.directive.ts [hunk 3]
```diff
@@ -0,0 +1,207 @@
+import { ng, Behaviours ,workspace} from 'entcore';
```
Comment (Rseuret) [inline src/main/resources/public/ts/directives/file-viewer/file-viewer.directive.ts:1]: directive ode + html séparé pour futur

File: src/main/resources/public/ts/controllers/board-view.controller.ts [hunk 4]
```diff
@@ -683,7 +683,8 @@ class Controller implements IViewModel {
         this.cardForm.caption = "";
         this.cardForm.resourceId = file._id;
         this.cardForm.resourceFileName = file.metadata.filename;
-
+        this.cardForm.resource = file;
+        console.log("onFileSelected")
```
Comment (flmartineau) [inline src/main/resources/public/ts/controllers/board-view.controller.ts:687]: console.log

File: src/main/resources/public/ts/directives/file-viewer/file-viewer.directive.ts [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+                _txtCache[scope.vm.file._id] = new CsvProviderFromText(scope.vm.file);
+                return _txtCache[scope.vm.file._id];
+            }
+
+            if (scope.contentType == 'html') {
+                const call = async () => {
+                    const a = await workspaceService.getDocumentBlob(scope.vm.file._id);
+                    const reader = new FileReader();
+                    reader.onload = function () {
+                        // scope.htmlContent = $sce.trustAsHtml(reader.result) as string;
```
Comment (flmartineau) [inline src/main/resources/public/ts/directives/file-viewer/file-viewer.directive.ts:196]: Enleve le commentaire

**Review by flmartineau — APPROVED** [review]

**Review by flmartineau — APPROVED** [review]

**Review by FlorentMr — APPROVED** [review]

**Pull request merged**